vortex-array/src/array/primitive/mod.rs

```diff
@@ -60,7 +61,7 @@ impl PrimitiveArray {
                     .to_metadata(length)
                     .vortex_expect("Invalid validity"),
             }),
-            Some(buffer),
+            Some(ArrayBuffer::new_with_alignment(buffer, ptype.byte_width())),
```

Comment:
should constructor just accept ArrayBuffer directly?

Reply:
It should accept a ScalarBuffer<T>, and then the PType can be extracted. For now it's a bit annoying to take an ArrayBuffer and then make sure alignment matches when I'm about to change it anyway